packages/core/src/external/carequality/ihe-gateway-v2/outbound/__tests__/process-dr.test.ts

```diff
@@ -1,12 +1,123 @@
+    });
+  });
+
+  testFiles.forEach(({ name, mimeType, fileExtension }) => {
+    it(`[mtom with attachments]: should process the ${fileExtension} DR response correctly`, async () => {
+      const xmlTemplatePath = path.join(__dirname, "./xmls/dr-no-mime-type.xml");
+      const xmlTemplate = fs.readFileSync(xmlTemplatePath, "utf8");
+
+      const fileContent = fs.readFileSync(path.join(__dirname, `./files/${name}`));
+      const fileContentB64 = fileContent.toString("base64");
+      const modifiedXml = xmlTemplate.replace(
+        "<Document></Document>",
+        `<Document>${fileContentB64}</Document>`
+      );
+
+      const mtomResponse = await createMtomMessageWithoutAttachments(modifiedXml);
+      const response = await processDrResponse({
+        drResponse: {
+          mtomResponse: mtomResponse,
+          gateway: outboundDrRequest.gateway,
+          outboundRequest: {
+            ...outboundDrRequest,
+            documentReference: outboundDrRequest.documentReference.map(docRef => ({
+              ...docRef,
+              metriportId: uuidv4(),
+            })),
+          },
+        },
+      });
+      expect(response?.documentReference?.[0]?.contentType).toEqual(mimeType);
+    });
+  });
+
+  it("[mtom with multiple attachments]: should correctly create and parse MTOM message with XML, PDF, and TIFF attachments", async () => {
+    const xmlFile = testFiles.find(file => file.fileExtension === ".xml");
+    const pdfFile = testFiles.find(file => file.fileExtension === ".pdf");
+    const tiffFile = testFiles.find(file => file.fileExtension === ".tiff");
+
+    if (!xmlFile || !pdfFile || !tiffFile) {
+      throw new Error("Required test files not found");
+    }
+
+    const xmlFilePath = path.join(__dirname, "./files/", xmlFile.name);
+    const pdfFilePath = path.join(__dirname, "./files/", pdfFile.name);
+    const tiffFilePath = path.join(__dirname, "./files/", tiffFile.name);
+
+    const xmlBuffer = fs.readFileSync(xmlFilePath);
+    const pdfBuffer = fs.readFileSync(pdfFilePath);
+    const tiffBuffer = fs.readFileSync(tiffFilePath);
+
+    const attachmentsData = [
+      { payload: xmlBuffer, mimeType: xmlFile.mimeType },
+      { payload: pdfBuffer, mimeType: pdfFile.mimeType },
+      { payload: tiffBuffer, mimeType: tiffFile.mimeType },
+    ];
+
+    const mtomAttachments = await createMtomMessageWithAttachments(attachmentsData);
+    const response = await processDrResponse({
+      drResponse: {
+        mtomResponse: mtomAttachments,
+        gateway: outboundDrRequestMtom.gateway,
+        outboundRequest: outboundDrRequestMtom,
+      },
+    });
+
+    expect(response).toBeDefined();
+    expect(response.documentReference).toHaveLength(3);
+    expect(response.documentReference?.[0]?.contentType).toBe(xmlFile.mimeType);
+    expect(response.documentReference?.[1]?.contentType).toBe(pdfFile.mimeType);
+    expect(response.documentReference?.[2]?.contentType).toBe(tiffFile.mimeType);
```

Comment:
nit: this requires files on a certain order; while it works w/ the current impl, if it its not a requirement/contract one could build the test making it so `response` contains doc refs w/ the expected contents; something like:

```ts
expect(response.documentReference).toBeTruthy();
expect(response.documentReference).toHaveLength(3);
const contentTypes = response.documentReference?.map(d => d.contentType);
expect(contentTypes).toEqual(
  expect.arrayContaining([xmlFile.mimeType, pdfFile.mimeType, tiffFile.mimeType])
);
expect(response.documentReference?.[0]?.contentType).toBe(xmlFile.mimeType);
```